### ISSUE TITLE
Connecting to hidden tor service

### ISSUE THREAD

**Issue by zoof** [jlund/streisand#791]
### Expected behavior:
Connects
### Actual Behavior:
Does not connect
### Steps to Reproduce:
1. Try to connect

### Additional Details:

07/06/2017 23:31:01.400 [NOTICE] Bootstrapped 5%: Connecting to directory server 
07/06/2017 23:31:01.400 [NOTICE] Bootstrapped 10%: Finishing handshake with directory server 
07/06/2017 23:31:01.500 [WARN] Proxy Client: unable to connect to xx.xx.xx.xx:9443 ("general SOCKS server failure") 
07/06/2017 23:31:03.300 [NOTICE] Closing no-longer-configured Socks listener on 127.0.0.1:9150 
07/06/2017 23:31:03.300 [NOTICE] DisableNetwork is set. Tor will not make or accept non-control network connections. Shutting down all existing connections. 
07/06/2017 23:31:03.300 [NOTICE] Closing old Socks listener on 127.0.0.1:9150 

In looking at the torrc files, it appears that the old torrc file was replaced with a typical torrc file that contains only commented suggested settings: https://pastebin.com/Rm2n24Eg

##### *Target Cloud Provider:*
GCE
##### *Operating System of target host:*
Ubuntu 16.04
##### *Operating System of client:*
Debian Jessie

**Comment by cpu** [comment]
@zoof Was this an existing Streisand server or one that you've recently provisioned?

**Comment by zoof** [comment]
This was an existing server.

**Comment by cpu** [comment]
> This was an existing server.

What is the output from `tor --version`? Have you made any modifications to the unattended upgrades configuration in `/etc/apt/apt.conf/`? 

Based on my understanding of dpkg/apt it seems very strange that the modified Streisand torrc would be replaced by the distro packaged version without explicitly asking you to make a choice about which to keep in a conffile prompt. Any theories from your side that might explain that?

**Comment by zoof** [comment]
# tor --version
    Tor version 0.3.0.9 (git-100816d92ab5664d).

I have not made any changes to apt.conf but it sounds probable that I ran `apt upgrade` and answered yes to install the package maintainer's version of torrc.  The fix seems simple enough -- `sudo cp /etc/tor/torrc.dpkg-old /etc/tor/torrc`.

Is it the case that I should refrain from manually updating anything and just let the system do so on its own?

**Comment by cpu** [comment]
Hi @zoof,

> I have not made any changes to apt.conf but it sounds probable that I ran apt upgrade and answered yes to install the package maintainer's version of torrc. The fix seems simple enough -- sudo cp /etc/tor/torrc.dpkg-old /etc/tor/torrc.

I noticed that the tor upgrade was stalled on a distfile prompt on one of my own servers that I manually resolved (making the opposite choice you did).

After noticing that I submitted a PR (https://github.com/jlund/streisand/pull/781) to update our unattended upgrades script to tell dpkg to always keep the Streisand version of config files automatically without prompting.

> Is it the case that I should refrain from manually updating anything and just let the system do so on its own?

With the commit from #781 applied the automatic security updates should do the right thing on its own. In general I tell folks that we support security updates but not feature updates. It's possible if you install non-security updates on the server yourself that it will break something. I'm hopeful we can get into a position where we can aim to support a fully-up-to-date 16.04 system in the future.

Hope that helps!
